### PR TITLE
apple arm64 support

### PR DESCRIPTION
**Pull request by ScanMountGoat**
This adds precompiled kernels for arm64 on MacOS to allow using the crate without an ISPC compiler on Apple silicon. I also needed to remove the target ISA configuration to avoid requiring x86 instructions. It looks like not specifying the target ISA uses the native ISA when compililing from the [docs](https://docs.rs/ispc_compile/1.1.0/ispc_compile/opt/enum.TargetISA.html). The previous code specifies almost all of the x86 options, so I'm not sure if using the native ISA will cause any issues. Let me know if there's a better way of handling this.

### CODE REVIEW

**Comment by MarijnS95** [comment]
Thanks! Would you mind adding these (cross-compiling) steps to the CI instead (and upload artifacts provided by it) similar to how we do this for Android (https://github.com/Traverse-Research/intel-tex-rs-2/commit/d8c2a1e4b244b523b6282122ba562b86b3be11170) so that we also have the capacity to rebuild/update the binaries on crate updates?

EDIT: I'll submit the changes to cross-compile for Android/Linux aarch64 soon after #11 lands :)

> The previous code specifies almost all of the x86 options, so I'm not sure if using the native ISA will cause any issues. Let me know if there's a better way of handling this.

I recall switching to `NEON` for our `ispc-downsampler` crate (https://github.com/Traverse-Research/ispc-downsampler/pull/5/files#diff-d0d98998092552a1d3259338c2c71e118a5b8343dd4703c0c7f552ada7f9cb42) since "native ISA" has no definition when cross-compiling.

**Comment by ScanMountGoat** [comment]
> Thanks! Would you mind adding these (cross-compiling) steps to the CI instead (and upload artifacts provided by it) similar to how we do this for Android ([d8c2a1e](https://github.com/Traverse-Research/intel-tex-rs-2/commit/d8c2a1e4b244b523b6282122ba562b86b3be11170)) so that we also have the capacity to rebuild/update the binaries on crate updates?

I've added the steps to CI. Just to clarify, you just want the changes to the CI script and don't want my commits to include the Apple silicon binaries? It's not a big deal to rework the commits on my end either way.

**Comment by MarijnS95** [comment]
Hmm unfortunately I cannot trigger the workflow run on branches from forks, but you _might_ need to install the cross-compilation targets `rustup target add ...` before this starts working.

> I've added the steps to CI. Just to clarify, you just want the changes to the CI script and don't want my commits to include the Apple silicon binaries? It's not a big deal to rework the commits on my end either way.

In reality yes, I'd prefer if the compiled binaries come from the CI, but we haven't set up a workflow for this yet (so it's 1. download from GitHub 2. Unpack and commit 3. Push to PR) which is inconvenient regardless.

**Comment by MarijnS95** [comment]
Hmm `clippy` is also complaining about `uninlined_format_args` even though that lint should have been demoted to `pedantic` in the recent 1.67 release...

**Comment by MarijnS95** [comment]
Should be addressed now if you rebase :)

File: .github/workflows/generate-binaries.yaml
Comment (MarijnS95) [inline .github/workflows/generate-binaries.yaml]: Fwiw I had some trouble (https://github.com/Traverse-Research/intel-tex-rs-2/commit/fa09a41929b3484772d14e6a634b7955c77601eb)
getting `{}` in globs to work and https://github.com/actions/toolkit/tree/main/packages/glob#patterns doesn't state support for this, we'll have to write them out :(

File: .github/workflows/generate-binaries.yaml
Comment (MarijnS95) [inline .github/workflows/generate-binaries.yaml:60]: Just like d8c2a1e4b244b523b6282122ba562b86b3be1117 you probably have to add a `rustup target add ...` step above.

**Review by MarijnS95 — COMMENTED** [review]
Looks mostly alright to me, let's re-evaluate once d8c2a1e4b244b523b6282122ba562b86b3be1117 is in which I intend to just merge in soon.

**Comment by ScanMountGoat** [comment]
I've update my fork with the suggested changes. I'm not including the precompiled kernels for now since you mentioned wanting to use the upload action for that.

**Comment by MarijnS95** [comment]
Thanks @ScanMountGoat! I've since opened #17 for the promised `aarch64` support on Linux/Android, and temporarily rebased your branch on it to [generate the binaries](https://github.com/Traverse-Research/intel-tex-rs-2/actions/runs/4388918712).

As mentioned above we cannot rely on a "native" ISA so I've set something I think is sensible, but it may not work for every device. I recall however not being allowed to compile for multiple `NEON` ISAs at once, something that is possible for x86 (where we have a mix of SSE and AVX revisions).

Will merge this after #17 and then we'll see how it goes out in the wild!

**Comment by ScanMountGoat** [comment]
Nice! Thanks for the update.

**Review by MarijnS95 — APPROVED** [review]

**Pull request merged**
